### DISCUSSION TITLE
Upgrade

### DISCUSSION THREAD

**Discussion by mfraser** [Guilouz/Creality-Helper-Script-Wiki#649] [General]
I'm still running V4.0.0 of the installer script with firmware V1.3.2.20 on a K1 Max and it is working fine, but I'm wondering if I'm missing much by not upgrading. 

Am I right in thinking that I'd have to follow the firmware upgrade procedure first as the helper script requires V1.3.3.5 or newer?

**Comment by jglerner** [comment]
@mfraser,

Use this to factory reset:

[https://guilouz.github.io/Creality-Helper-Script-Wiki/firmwares/reset-factory-settings/?h=factory+rese](url)

Then update your firmware. The last version is 1.3.3.29 and it's a safe update.

[https://guilouz.github.io/Creality-Helper-Script-Wiki/firmwares/install-and-update-rooted-firmware-k1/](url)

After that install the Newest Creality Helper Script:

[https://guilouz.github.io/Creality-Helper-Script-Wiki/helper-script/helper-script-installation/](url)

Finally, be happy!

PS: This procedure can be bumpy but is a safe path, do it at your own pace.

**Reply by jglerner** [reply]
@mfraser,

And don't forget about rooting the firmware before installing  Creality Halper Script!

**Reply by mfraser** [reply]
Thanks for this. One more question, does doing a factory reset remove all gcode files from the printer and if so how do I back them up?
